safe_transaction_service/history/models.py

```diff
@@ -1538,11 +1538,27 @@ class Meta:
         ordering = ["tx_block_number"]
 
 
+class SafeContractManager(models.Manager):
+    def get_banned_safes(self) -> QuerySet[ChecksumAddress]:
+        return self.filter(banned=True).values_list("address", flat=True)
+
+
 class SafeContract(models.Model):
+    objects = SafeContractManager()
     address = EthereumAddressV2Field(primary_key=True)
     ethereum_tx = models.ForeignKey(
         EthereumTx, on_delete=models.CASCADE, related_name="safe_contracts"
     )
+    # Avoid to index events from problematic safes like non verified contracts
+    banned = models.BooleanField(default=False)
+
+    indexes = [
+        Index(
+            name="banned_idx",
```

Comment:
```suggestion
            name="history_safe_banned_idx",
```

`history_safe_contract_banned_idx` would be better but Django has a limit of `30 chars` for index names